account_invoice_comment_template/migrations/13.0.1.0.0/post-migration.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+from openupgradelib import openupgrade
+
+drop_columns = [
+    ("account_move", "old_comment_template1_ids"),
+    ("account_move", "old_comment_template2_ids"),
+]
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        INSERT INTO account_move_base_comment_template_rel
+        (base_comment_template_id, account_move_id)
+        SELECT bct.id AS base_comment_template_id, am.id AS account_move_id
+        FROM account_move am
+        JOIN base_comment_template bct ON bct.id = am.old_comment_template1_ids
+        WHERE am.old_comment_template1_ids IS NOT NULL
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        INSERT INTO account_move_base_comment_template_rel
+        (base_comment_template_id, account_move_id)
+        SELECT bct.id AS base_comment_template_id, am.id AS account_move_id
+        FROM account_move am
+        JOIN base_comment_template bct ON bct.id = am.old_comment_template2_ids
+        WHERE am.old_comment_template2_ids IS NOT NULL
+        """,
+    )
+    openupgrade.drop_columns(env, drop_columns)
```

Comment:
Never drop columns. Remember: preservation principle. The only exception is if that data compromises integrity, which violates other principle. Even more, you will get an error, as they are m2m fields, not SQL columns. You can lift constraints in such tables for avoiding performance lost though.

Reply:
Sorry for my mistake, the name of the fields was wrong (it is not an m2m, it is 2 m2o fields that convert to m2m).

Reply:
You are still dropping columns and this is not the mentioned exception.